### PR TITLE
[RFR] Pass extra props down to root components

### PR DESCRIPTION
**Pull request by djhi**
- [x] Auth components
- [x] Fields
- [x] Views (Edit, Show, etc.)
- [x] Forms
- [x] Inputs
- [x] Document `formClassName`

### CODE REVIEW

**Comment by fzaninotto** [comment]
Argh! needs rebase.

File: packages/react-admin/src/mui/form/FormInput.js [hunk 1]
```diff
@@ -23,10 +23,7 @@ export const FormInput = ({ className, classes, input, ...rest }) =>
                     {input}
                 </Labeled>
             ) : (
-                React.cloneElement(input, {
-                    classes: { root: classes.input },
```
Comment (fzaninotto) [inline packages/react-admin/src/mui/form/FormInput.js]: why was it removed?

File: packages/react-admin/src/mui/input/FileInputPreview.js [hunk 2]
```diff
@@ -1,20 +1,14 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import classNames from 'classnames';
 import IconButton from 'material-ui/IconButton';
 import RemoveCircle from 'material-ui-icons/RemoveCircle';
 import { withStyles } from 'material-ui/styles';
 
 const styles = theme => ({
-    removeButtonHovered: {
```
Comment (fzaninotto) [inline packages/react-admin/src/mui/input/FileInputPreview.js]: I don't understand why you changed that

File: packages/react-admin/src/mui/input/ImageInput.js [hunk 3]
```diff
@@ -6,18 +6,28 @@ import translate from '../../i18n/translate';
...
 const styles = {
+    root: {},
     dropZone: {
         background: '#efefef',
         cursor: 'pointer',
         padding: '1rem',
         textAlign: 'center',
         color: '#999',
     },
-    preview: {
```
Comment (fzaninotto) [inline packages/react-admin/src/mui/input/ImageInput.js]: why do you change that?

File: packages/react-admin/src/mui/detail/Create.js [hunk 4]
```diff
@@ -156,10 +162,6 @@ Create.propTypes = {
     hasList: PropTypes.bool,
 };
 
-Create.defaultProps = {
```
Comment (fzaninotto) [inline packages/react-admin/src/mui/detail/Create.js]: Why did you remove that?

**Comment by fzaninotto** [comment]
I loved the `sanitizeRestProps` function that you used in one of the fileS. IMO you should use it more.

File: packages/react-admin/src/mui/input/FileInputPreview.js (hunk 2 above)
Comment (djhi) [reply packages/react-admin/src/mui/input/FileInputPreview.js]: Checkout the `ImageInput` component, I added JSS styles to handle mouse hover

File: packages/react-admin/src/mui/input/ImageInput.js (hunk 3 above)
Comment (djhi) [reply packages/react-admin/src/mui/input/ImageInput.js]: Because it's not needed anymore

File: packages/react-admin/src/mui/detail/Create.js (hunk 4 above)
Comment (djhi) [reply packages/react-admin/src/mui/detail/Create.js]: Because it's not used anywhere in this component, unlike in the `Edit` component

File: packages/react-admin/src/mui/input/FileInputPreview.js (hunk 2 above)
Comment (djhi) [reply packages/react-admin/src/mui/input/FileInputPreview.js]: This is what it looks like with those modifications screen

**Comment by fzaninotto** [comment]
needs rebase

**Comment by fzaninotto** [comment]
tests fail...

**Comment by fzaninotto** [comment]
missing documentation for `formClassName`

**Comment by oliviertassinari** [comment]
Great! I was looking for such pull-request. I will wait for the next release to clear the warnings :).
